### PR TITLE
Android bg location

### PR DESCRIPTION
**Pull request by amj311**
- Moves the Location Tracking logic from the useLocationHook and into Android native services.
- Location tracking and snitches happen even if the app is closed.
- Adds a NativeModule Service and Interface for abstracting interactions with native services (this should be used for later iOS adoption)

### CODE REVIEW

File: react-native-app/MainActivity.tsx [hunk 1]
```diff
@@ -2,11 +2,15 @@ import React from 'react';
...
 import config from './aws-exports';
 import Amplify from '@aws-amplify/core';
+import { NativeInput } from './models/NativeInput';
 
 Amplify.configure(config);
 
-export default function App(){
+
+export default function App(props){
+  console.log("*****FIT INITIAL PROPS", props)
```
Comment (connercharles) [inline react-native-app/MainActivity.tsx:11]: Do we want to leave these logs?

File: react-native-app/android/app/src/main/java/com/fitsnitchapp/LocationForegroundService.java [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+//                LocationUpdate newLocation = mGson.fromJson(
+//                        intent.getStringExtra(LOCATION_EVENT_DATA_NAME), LocationUpdate.class);
+
+
+            }
+        };
+    }
+
+
+//    private void sendLocationToJs(LocationUpdate location) {
```
Comment (connercharles) [inline react-native-app/android/app/src/main/java/com/fitsnitchapp/LocationForegroundService.java:90]: Curious, what were these all for?

**Review by connercharles — APPROVED** [review]
Holy crap this is a lot of stuff! You really had to dive into all that java too--gross. Good job, this is really impressive!

File: react-native-app/android/app/src/main/java/com/fitsnitchapp/LocationForegroundService.java (hunk 2 above)
Comment (amj311) [reply react-native-app/android/app/src/main/java/com/fitsnitchapp/LocationForegroundService.java:90]: Looks like the ruins of old strategies that didn't end up working. Guess I have some cleanup to do haha

File: react-native-app/MainActivity.tsx (hunk 1 above)
Comment (amj311) [reply react-native-app/MainActivity.tsx:11]: I think we should leave them in for now but start looking for better logging solutions. I'd like to have some sort of temporary log file that can be sent to us with bug reports

**Pull request merged**